CHANGELOG.md

```diff
@@ -1,6 +1,6 @@
 # Change log
 
-## Unreleased
+## check_patroni 2.0.0 - 2024-04-09
```

Comment:
Looking at the changelog, I don't see any breaking change, is there any?

If not, bumping the major version is not appropriate if you mean to follow semantic versioning (which I strongly recommend). And so `1.1.0` is probably more appropriate.

Reply:
I thought thoses might warrant a major version change.
Do you think it's a mistake ? Should this be made clearer in the `CHANGELOG.md` file ? 

> Add the timeline in the  `cluster_has_replica` perfstats. (#50 
> Add the leader type and whether it's archiving to the `cluster_has_leader` perfstats. (#58)

Modifying perf stats could have an impact depending on what users do with them ?

> Fix what `cluster_has_replica` deems a healthy replica.

Because of the change in node states made in Patroni,  what we deemed a healthy replica could no longer be one.

Reply:
If those changes are *breaking* changes, yes, this should be made very clear in the changelog and indeed the major version number must be bumped.

But are they really? I mean, would something work differently as before because you intentionally changed it (fix bugs does not count here)?

The current wording of these change items you mention does not make that clear to me: first two items state that you're adding things, the third mentions a fix. So as a user, I'd read this and think: okay, fine, no breaking change, let's upgrade.

Reply:
Hm; just realized that you already pushed the `v2.0.0` tag. Will be more difficult to sort out if it appears needed:  https://git-scm.com/docs/git-tag#_on_re_tagging

Reply:
Sorry for the long delay, I was on a client mission.

Yes, I thought it was a "package" with the MR...

Reply:
It could break things depending on how they use the perf data (which I am less convinced now) or what they expect to be a healthy replica.

Would it be better with this at the top ?

```
### Notice

While fixing `cluster_has_replica`, the definition of a healthy replica was changed. It's more restrictive now, hence the jump from v1 to v2.
```

Reply:
Yes; looks good.